src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -409,24 +409,29 @@ def on_request(self, request):
 
         # 'request_data' has been dumped into JSON string in set_json_body() when building HttpRequest in Python SDK.
         # In order to facilitate subsequent parsing, it is converted into a dict first
-        http_request.data = json.loads(request_data)
+        modified_data = json.loads(request_data)
```

Comment:
I would like to confirm that this is just replacing the variable from `http_request.data` to `modified_data` to improve the readability of the code, but there is no actual logical modification, right?
If so, could you please rerecord the `DeploymentTest` tests in live mode? Because the modified code here is too core and versatile, I suggest that we conduct do live tests to prevent unexpected situations

Reply:
This was deliberate - there **is a logical modification** - to fix a bug. The earlier logic was essentially doing this:

```python
http_request.data = json.loads(request_data)
if http_request.data.get('properties', {}).get('template'):
    ...
    http_request.data = http_request.data.encode('utf-8')
```

The bug that this PR is fixing is that if the `if` condition doesn't match (e.g. if there's no `.properties.template`), `http_request.data` is left in an invalid state.

My fix is essentially making sure that we only modify `http_request.data` inside the `if` condition, and only with valid properties.

Reply:
If you review the changes, you should find that this is **the only** logical change (aside from adding logging).

Everything else is a refactor, to use intermediate variables instead of using `http_request.data` as if it were an intermediate variable.

Reply:
I've re-run `azdev test --live DeploymentTest` and checked in the result.

Reply:
Make sense! Thanks for your careful and thoughtful contribution